admin_manual/ai/app_llm2.rst

```diff
@@ -42,6 +47,42 @@ This app allows supplying alternate LLM models as *gguf* files in the ``/nc_app_
 3. Restart the llm2 ExApp
 4. Select the new model in the Nextcloud AI admin settings
 
+
+Configuring alternate models
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since every model requires slightly different inference parameters, you can pass along a configuration file for the alternate model files you supply.
+
+The configuration file for a model file must have the same name as the model file but must end in ``.json`` instead of ``.gguf``.
+
+The strings ``{system_prompt}`` and ``{user_prompt}`` are variables that will be filled in by the app, so they must be part of your prompt template.
+
+Here is an example config file for Llama 2:
+
+.. code-block:: json
+
+   {
+     "prompt": "<|im_start|> system\n{system_prompt}\n<|im_end|>\n<|im_start|> user\n{user_prompt}\n<|im_end|>\n<|im_start|> assistant\n",
+     "gpt4all_config": {
+       "max_tokens": 4096,
+       "n_predict": 2048,
+       "stop": ["<|im_end|>"]
+     }
+   }
+
+Here is an example configuration for Llama 3:
+
+.. code-block:: json
+
+   {
+     "prompt": "<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n{system_prompt}<|eot_id|><|start_header_id|>user<|end_header_id|>\n{user_prompt}<|eot_id|>\n<|start_header_id|>assistant<|end_header_id|>\n",
```

Comment:
it's two `\n`s after `<|end_header_id|>`. Doesn't affect the output much but would be better still.
https://llama.meta.com/docs/model-cards-and-prompt-formats/meta-llama-3/

Reply:
The notes on that page say: "Newlines are part of the prompt format, for clarity in the examples, they have been represented as actual new lines", so I believe, every empty line in their examples, is *on* `\n`

Reply:
hmm, it says that but see here in their repo they use the double `\n`: https://github.com/meta-llama/llama3/blob/main/llama/tokenizer.py#L206-L212